rest/server_context.go

```diff
@@ -41,18 +41,19 @@ const KDefaultNumShards = 16
 // This struct is accessed from HTTP handlers running on multiple goroutines, so it needs to
 // be thread-safe.
 type ServerContext struct {
-	config            *StartupConfig
-	persistentConfig  bool
-	bucketDbName      map[string]string              // bucketDbName is a map of bucket to database name
-	dbConfigs         map[string]*DatabaseConfig     // dbConfigs is a map of db name to DatabaseConfig
-	databases_        map[string]*db.DatabaseContext // databases_ is a map of dbname to db.DatabaseContext
-	lock              sync.RWMutex
-	statsContext      *statsContext
-	bootstrapContext  *bootstrapContext
-	HTTPClient        *http.Client
-	cpuPprofFileMutex sync.Mutex     // Protect cpuPprofFile from concurrent Start and Stop CPU profiling requests
-	cpuPprofFile      *os.File       // An open file descriptor holds the reference during CPU profiling
-	_httpServers      []*http.Server // A list of HTTP servers running under the ServerContext
+	config               *StartupConfig
+	initialStartupConfig *StartupConfig
```

Comment:
Now we have 2 configs here, might make sense to add comments.

```suggestion
	config               *StartupConfig // The current runtime configuration of the server
	initialStartupConfig *StartupConfig // The configuration at startup of the server
```